### ISSUE TITLE
MPM monitoring hangs, miner continues (MPH)

### ISSUE THREAD

**Issue by P9HKU** [MultiPoolMiner/MultiPoolMiner#1245]
Apologies if silly question, new to mining & MPM.

MPM is running fine, with MiningPoolHub. Randomly, although usually within 24 hours of starting MPM, the 'monitoring' window freezes.  That is to say, it stops updating. So benches, current status, profitability, historical tasks - all static.  This does not rectify other than by closing & restarting MPM.

The miner continues to run, with normal hashrates, and my MPH balances continue to rise at roughly the expected rates.

Here is the last entry in my log (user & rig name redacted). (Other than the PS>TerminagingError... line, this is exactly what the 'monitoring' window has shown for 24 hours now. The DSTM miner is still churning away for me.)

Miner            Algorithm       Speed BTC/Day Accuracy BTC/GH/Day Pool
-----            ---------       ----- ------- -------- ---------- ----
Dstm             Equihash    1.71 KH/s 0.00124      90%  728.74886 MiningPoolHub-Zclassic
Ewbf             Equihash    1.63 KH/s 0.00119      90%  728.74886 MiningPoolHub-Zclassic
ExcavatorNvidia1 Equihash    1.63 KH/s 0.00119      90%  728.74886 MiningPoolHub-Zclassic
ExcavatorNvidia2 Equihash    1.60 KH/s 0.00117      90%  728.74886 MiningPoolHub-Zclassic
ExcavatorNvidia2 Lyra2RE2  149.09 MH/s 0.00087      88%    0.00581 MiningPoolHub-Vertcoin
ExcavatorNvidia1 Lyra2RE2  146.14 MH/s 0.00085      88%    0.00581 MiningPoolHub-Vertcoin
CcminerLyra2RE2  Lyra2RE2  138.16 MH/s 0.00080      88%    0.00581 MiningPoolHub-Vertcoin
CcminerTpruvot   Lyra2z      5.32 MH/s 0.00075      99%    0.14097 MiningPoolHub-Zcoin
CcminerTpruvot   Lyra2RE2  124.83 MH/s 0.00073      88%    0.00581 MiningPoolHub-Vertcoin
CcminerNanashi   Lyra2z      4.75 MH/s 0.00067      99%    0.14097 MiningPoolHub-Zcoin
CcminerTpruvot   Equihash  340.00  H/s 0.00025      90%  728.74886 MiningPoolHub-Zclassic

   Type: {AMD, NVIDIA}

Miner      Algorithm     Speed BTC/Day Accuracy BTC/GH/Day Pool
-----      ---------     ----- ------- -------- ---------- ----
Excavator1 Equihash  1.63 KH/s 0.00119      90%  728.74886 MiningPoolHub-Zclassic
Excavator2 Equihash  1.60 KH/s 0.00117      90%  728.74886 MiningPoolHub-Zclassic

   Status: Running

    Speed Active                      Launched Command
    ----- ------                      -------- -------
1.70 KH/s 00 Days 12 Hours 56 Minutes 8 Times  Bin\Equihash-DSTM\zm.exe --telemetry --server europe.equihash-hub.miningpoolhub.com --port 17023 --user xxx.xxx --pass x

   Status: Idle

 Speed Active                      Launched Command
 ----- ------                      -------- -------
0  H/s 00 Days 00 Hours 19 Minutes Once     Bin\Equihash-EWBF\miner.exe --eexit 1 --api --server europe.equihash-hub.miningpoolhub.com --port 17023 --user xxx.xxx --pass x
                                            --fee 0 --intensity 64
0  H/s 00 Days 00 Hours 22 Minutes Once     Bin\Excavator\excavator.exe -p 23456 -c MiningPoolHub_Equihash_xxx.xxx_2_Nvidia.json -na
0  H/s 00 Days 00 Hours 30 Minutes Once     Bin\Excavator\excavator.exe -p 23456 -c MiningPoolHub_Lyra2RE2_xxx.xxx_2_Nvidia.json -na
0  H/s 00 Days 00 Hours 15 Minutes 2 Times  Bin\Equihash-DSTM\zm.exe --telemetry --server europe.equihash-hub.miningpoolhub.com --port 17023 --user aaronsace.multipoolminer
                                            --pass x

Miner Pool          Algorithm Watchdog Timer
----- ----          --------- --------------
Dstm  MiningPoolHub Equihash       0 Seconds

PS>TerminatingError(Invoke-RestMethod): "A task was canceled."
WARNING: Pool API (AHashPool) has failed.

-----------------------------------
My concern is that while the rig continues to mine (yay!) it is not updating profitability and therefore not switching accordingly (booooo) and if that's the case, then the devs are not getting their 15 minutes either (boooooo??? lol)

Anyone else had this, or know a possible cause?

Poh.

**Comment by UselessGuru** [comment]
> the 'monitoring' window

What window? Please post a screenshot.

[And could this be the reason?](https://github.com/MultiPoolMiner/MultiPoolMiner/issues/547)?
See AdrianAcala' answer

**Comment by P9HKU** [comment]
My apologies, I didn't explain that well.  Can't SS right now as away from home and can't RDP from outside. I mean the main MPM window, that shows benchmarks and current status, etc. (As opposed to the window showing the activity of the current miner.)

**Comment by platonicjesus** [comment]
I'm having the same issue but it only happens when I am running the cpu AND the AMD miner. If I just run the AMD miner it never hangs.

**Comment by P9HKU** [comment]
Well, it hasn't done it since I made this thread, albeit that's only a couple of days. So either:

Semi-random event, not yet replocated;
or
The fix that was linked above from AdrianAcala resolved the issue;
or
Removing all pools except MPH from the Pools folder resolved the issue.

If it happens again I will be more methodical in trying the fixes to try and get an answer.

**Comment by CptanPanic** [comment]
I haven't updated in a bit, but it has started doing this for me also. The real reason I know is because I am using the monitoring feature and it stops updating, and when I check the powershell window it is frozen. So I have to close it and restart it. The window looks normal but doesn't update.

**Comment by aaronsace** [comment]
@CptanPanic Please can you upload the logs?

**Comment by UselessGuru** [comment]
> The most critical looking thing is

Nope, these errors are not true errors and do not cause any problems.

Please post the MPM log (Logs\MultiPoolMiner_YYYY-MM-DD.log)

**Comment by mwodzynski** [comment]
It seems that I have the same problem, The monitoring window no longer updates, the miner is working. See screen shoots:

![screenshot 3](https://user-images.githubusercontent.com/34895306/36851347-a0d91914-1d2e-11e8-8b18-163b9f3c85f7.png)


and 

![screenshot 6](https://user-images.githubusercontent.com/34895306/36851366-ac91af46-1d2e-11e8-865f-7c14203cf7df.png)

Please let me know if you need any logs from me.

**Comment by mwodzynski** [comment]
@UselessGuru 

I just installed latest patches. It seems that the settings for that window where already marked as recommended in AdrianAcala's Issue. I will keep an eye on this and give you any updates.

**Comment by robl45** [comment]
Had same or similar problem last night, window just stopped updating at 5:56 or so.  Miner appeared to be still be running when I got up at 8:22

![image](https://user-images.githubusercontent.com/34346751/36946141-3e6f43a2-1f86-11e8-8d99-1bb5b8806f38.png)
